package/base-files/files/lib/upgrade/nand.sh

```diff
@@ -7,6 +7,12 @@
 # 'kernel' partition on NAND contains the kernel
 CI_KERNPART="${CI_KERNPART:-kernel}"
 
+# 'dtb' partition on NAND contains the dtb
+CI_DTBPART="${CI_DTBPART:-dtb}"
+
+# 'ramdisk' partition on NAND contains the ramdisk
+CI_RAMDISKPART="${CI_RAMDISKPART:-ramdisk}"
```

Comment:
The Ramdisk is fake - do you actually need to update this partition when performing sysupgrade?

Reply:
When I tried to leave the ramdisk as per factory image, it would fail to boot.
When I wiped the partition, it also failed to boot.
When I put a fake ramdisk, it allowed it to boot.
When I tried to change the uboot boot parameters so that it didn't need the ramdisk, it ignored the environment settings and still attempted to use the ramdisk.

Reply:
I don't know so much about that script, but again I wonder whether this has to be done for entire OpenWrt instead of just applying it on a more local basis.

Reply:
The script will only execute the portions of the nand if a) the image is present b) the mtd is labelled, but I understand your concern.
I can test the changes on other nand devices that I have, if you wish ?

Reply:
Changes to `package/base-files/files/lib/upgrade/nand.sh` have been full dropped.